src/pages/static/js/tabs.js

```diff
@@ -0,0 +1,84 @@
+class TabsBox extends HTMLElement {
+    constructor() {
+      super();
+      this.attachShadow({ mode: 'open' });
+      this.render();
+      this.loadTabs();
+    }
+  
+    async loadTabs() {
+      try {
+        if (!window.electronAPI || !window.electronAPI.getTabs) {
+          throw new Error('electronAPI not available');
+        }
+        const data = await window.electronAPI.getTabs();
+        const parsed = data ? JSON.parse(data) : null;
+        this.displayTabs(parsed);
+      } catch (e) {
+        console.error('Failed to load tabs', e);
+        this.displayTabs(null);
+      }
+    }
+  
+    displayTabs(tabsData) {
+      const container = this.shadowRoot.querySelector('.tabs-container');
+      container.innerHTML = '';
+  
+      if (!tabsData || !tabsData.tabs || tabsData.tabs.length === 0) {
+        container.innerHTML = '<p>No tabs found.</p>';
+        return;
+      }
+  
+      tabsData.tabs.forEach(tab => {
+        const item = document.createElement('div');
+        item.className = 'tab-item';
+        item.innerHTML = `
+          <span class="title">${tab.title || tab.url}</span>
+          <div class="actions">
+            <button class="activate-btn" data-id="${tab.id}">Switch</button>
```

Comment:
We don't need a switch button; it should open the tab when the title is clicked.